File: Computacion_Grafica/lab8/traslacion.py
from OpenGL.GL import *
from OpenGL.GLUT import *
from OpenGL.GLU import *

def Traslada():
    # Triangulo antes de trasladar
    glBegin(GL_TRIANGLES)
    glColor3f (0.5 , 0.3 , 0.9) # Color1
    glVertex2f(100.0, 100.0)
    glVertex2f(200.0, 100.0)
    glVertex2f(150.0, 150.0)
    glEnd()

    # Triangulo después de trasladar
    glTranslatef(200.0, 200.0, 0.0)
    glBegin(GL_TRIANGLES)
    glColor3f(0.8, 0.26, 1.0) # Color2
    glVertex3f(100.0, 100.0, 0.0)
    glVertex3f(200.0, 100.0, 0.0)
    glVertex3f(150.0, 150.0, 0.0)
    glEnd()

def Rota():
    #--triangulo antes de rotar
    glColor3f (0.5 , 0.3 , 0.9) # Color1
    glBegin(GL_TRIANGLES)
    glVertex2f(100.0, 100.0)
    glVertex2f(200.0, 100.0)
    glVertex2f(150.0, 150.0)
    glEnd()

    #--triangulo rotado
    glRotatef(45, 0, 0, 1)
    glBegin(GL_TRIANGLES)
    glColor3f(0.8, 0.26, 1.0) # Color2
    glVertex2f(100.0, 100.0)
    glVertex2f(200.0, 100.0)
    glVertex2f(150.0, 150.0)
    glEnd()
def Escala():
    #--triangulo antes de escalar
    glColor3f (0.5 , 0.3 , 0.9) # Color1
    glBegin(GL_TRIANGLES)
    glVertex2f(100.0, 100.0)
    glVertex2f(200.0, 100.0)
    glVertex2f(150.0, 150.0)
    glEnd()

    #--triangulo despues de escalar
    glScalef(2.0, 2.0, 2.0)
    glBegin(GL_TRIANGLES)
    glColor3f(0.8, 0.26, 1.0) # Color2
    glVertex2f(100.0, 100.0)
    glVertex2f(200.0, 100.0)
    glVertex2f(150.0, 150.0)
    glEnd()
    
def TransfComp():
    #--triangulo antes de rotar y trasladar
    glColor3f (0.5 , 0.3 , 0.9) # Color1
    glBegin(GL_TRIANGLES)
    glVertex2f(100.0, 100.0)
    glVertex2f(200.0, 100.0)
    glVertex2f(150.0, 150.0)
    glEnd()

    #--triangulo después de rotar y trasladar
    glRotatef(-20, 0, 0, 1)
    glTranslatef(100.0, 100.0, 0.0)
    glBegin(GL_TRIANGLES)
    glColor3f(0.8, 0.26, 1.0) # Color2
    glVertex2f(100.0, 100.0)
    glVertex2f(200.0, 100.0)
    glVertex2f(150.0, 150.0)
    glEnd()

    glFlush()
# Despliega el gráfico
def display():
    glClear(GL_COLOR_BUFFER_BIT)
    # Traslada()
    # Rota()
    # Escala()
    TransfComp()
    glFlush() 

def myinit():
    glClearColor (0.9 ,0.92 , 0.95 , 1.0) # Fondo
    glPointSize(1.0)
    glMatrixMode(GL_PROJECTION)
    glLoadIdentity() 
    gluOrtho2D(0.0, 499.0, 0.0, 499.0)

def main():
    # Inicializacion GLUT estándar
    glutInit()
    glutInitDisplayMode(GLUT_SINGLE | GLUT_RGB)
    glutInitWindowSize(500, 500) 
    glutInitWindowPosition(0, 0)
    glutCreateWindow("Transformaciones con OpenGL")
    glutDisplayFunc(display)
    myinit()
    glutMainLoop()

if __name__ == "__main__":
    main()

File: Computacion_Grafica/lab8/traslacion.cpp
#include<math.h>
#include<cstdlib>
#include<GL/glew.h>
#include<GL/glut.h>
#include<stdio.h>
using namespace std;

void Traslada()
{
    //--triangulo antes de trasladar
    glBegin(GL_TRIANGLES);
    glColor3f(0.5f, 1.0f, 0.7f);
    glVertex2f(100.0f, 100.0f);
    glVertex2f(200.0f, 100.0f);
    glVertex2f(150.0f, 150.0f);
    glEnd();

    //--triangulo después de trasladar
    glTranslatef(200.0f, 200.0f, 0.0f);
    glBegin(GL_TRIANGLES);
    glVertex3f(100.0f, 100.0f, 0.0f);
    glVertex3f(200.0f, 100.0f, 0.0f);
    glVertex3f(150.0f, 150.0f, 0.0f);
    glEnd();
}

//--despliega el gráfico
void display()
{
    glClear(GL_COLOR_BUFFER_BIT); //--establece el color de la ventana
    Traslada();    //--triangulo antes de trasladar
    glFlush(); //--fuerza la ejecución de los comandos de OpenGL
}

void myinit()
{
    glClearColor(1.0, 1.0, 1.0, 1.0);
    glColor3f(1.0, 0.0, 0.0);
    glPointSize(1.0); //--tamño de los puntos
    glMatrixMode(GL_PROJECTION);
    glLoadIdentity(); //--cargar la identidad y poder hacer las
    //-- transformaciones
    gluOrtho2D(0.0, 499.0, 0.0, 499.0); // Aquí está la corrección
}

int main(int argc, char** argv)
{
    /* Inicializacion GLUT estándar*/
    glutInit(&argc, argv);
    glutInitDisplayMode(GLUT_SINGLE | GLUT_RGB);
    glutInitWindowSize(500, 500); /* ventana 500x500 pixeles */
    glutInitWindowPosition(0, 0);
    glutCreateWindow("Transformaciones con OpenGL");
    glutDisplayFunc(display);
    myinit(); /* fija o establece los atributos */
    glutMainLoop(); /* entra a un ciclo de evento */
    return 0;
}

